src/instrumentation.ts

```diff
@@ -4,11 +4,18 @@ import indent from "./indent";
 import { Client } from "./pg";
 import { ParsedSettings } from "./settings";
 
-export async function runQueryWithErrorInstrumentation(
+interface InstrumentationError extends Error {
+  severity: string;
+  code: string;
+  detail: string;
+  hint: string;
```

Comment:
I don't think we can rely on any of these existing; also there's the `_gmLogged` hack we do:

```suggestion
  severity?: string;
  code?: string;
  detail?: string;
  hint?: string;
  _gmlogged?: boolean;
  _gmMessageOverride?: string;
```

Reply:
That `gmlogged` doesn't have a capital L pains me.

Reply:
is that something from inside migrate? can we change that?

Reply:
Yeah we could change it, I'm not sure what the effects would be...